src/pyfmi/fmi_util.pyx

```diff
@@ -310,7 +313,22 @@ cpdef prepare_data_info(np.ndarray[int, ndim=2] data_info, list sorted_vars, mod
 
     data_info[0,0] = 0; data_info[1, 0] = 1; data_info[2, 0] = 0; data_info[3, 0] = -1
 
-    data = np.append(np.append(np.append(model.time, model.get_real(param_real)),model.get_integer(param_int).astype(float)),model.get_boolean(param_bool).astype(float))
+    for i in range(nof_sorted_vars+1, nof_sorted_vars+1+nof_diag_params):
+        data_info[0,i] = 1
+        data_info[2,i] = 0
+        data_info[3,i] = -1
+        index_fixed = index_fixed + 1
+        data_info[1,i] = index_fixed
+
+    last_index = 0    
+    for i in range(nof_sorted_vars+1+nof_diag_params, nof_sorted_vars+1+nof_diag_params+nof_diag_vars):
+        data_info[0,i] = 3
+        data_info[2,i] = 0
+        data_info[3,i] = -1
+        last_index = last_index + 1
+        data_info[1,i] = last_index
+
+    data = np.append(np.append(np.append(np.append(model.time, model.get_real(param_real)),model.get_integer(param_int).astype(float)),model.get_boolean(param_bool).astype(float)), np.array(diagnostics_param_values).astype(float))
```

Comment:
Sorry, but why on earth is that many calls nestled like that?
I see that there was a chain of append calls before and you just added one more, but you should still clean up things like this when you need to change them.
Looking at the docs ([numpy.append from NumPy 1.21](https://numpy.org/doc/stable/reference/generated/numpy.append.html), I hope that is the correct one), append accepts a list for the second argument. As far as I can tell, just calling it with all except the first argument lumped into a list will give the same results. There is also an axis arguments that lets you add columns or rows to a matrix and similar. 
Do note that each nestled call to append creates a new copy of the entire array.